### ISSUE TITLE
Ruff rule suggestion: warn for useless f-strings

### ISSUE THREAD

**Issue by Datamine** [astral-sh/ruff#16246]
### Description

Occasionally I have seen f-strings used like this:
```
my_variable = "foo bar"
print(f"{my_variable}")

if f"{my_variable}" == "baz":
    return
```

In these instances, the f-string usage is improper because:

- The f-string contains only this variable, and nothing else. Using the f-string syntax is superfluous.
- If the user wanted to use a non-string variable as a string, then explicitly casting it using `str(my_variable)` would be more Pythonic.

I haven't seen a RUF rule to this effect, though I've seen a few other f-string rules. I think this would be a fine addition.

**Comment by MichaReiser** [comment]
Thanks for this rule suggestion. 

As you pointed out: Whether the use of `my_variable` is unnecessary depends on if `my_variable` is already a string. Ruff's capability to detect this right now are very limited and I worry that the rule would have too many false-negatives (it couldn't flag an f-string like that if it can't determine the type of `my_variable`. 

Related https://github.com/astral-sh/ruff/issues/16103

**Comment by merwok** [comment]
This could be opt-in to start?
